edx_rest_api_client/auth.py

```diff
@@ -48,3 +48,16 @@ def __call__(self, r):
         """ Update the request headers. """
         r.headers['Authorization'] = 'Bearer {}'.format(self.token)
         return r
+
+
+class OidcIdTokenAuth(AuthBase):
```

Comment:
i think we can just make a note of this in the docstring.  for client use within the context of a single HTTP request, i don't expiration is a worry.  For other cases, there's nothing preventing a consumer from passing un/key to get a fresh JWT every time, or manually managing the id token they pass in.